pkg/clusteragent/admission/mutate/tags_test.go

```diff
@@ -95,6 +95,75 @@ func Test_injectTagsFromLabels(t *testing.T) {
 	}
 }
 
+func Test_injectTags(t *testing.T) {
+	tests := []struct {
+		name        string
+		labels      map[string]string
+		pod         *corev1.Pod
+		wantPodFunc func() corev1.Pod
+	}{
+		{
+			name: "tag labels and injection on",
+			pod: withLabels(
+				fakePod("foo-pod"),
+				map[string]string{
+					"admission.datadoghq.com/enabled": "true",
+					"tags.datadoghq.com/env":          "dev",
+					"tags.datadoghq.com/service":      "dd-agent",
+					"tags.datadoghq.com/version":      "7",
+				},
+			),
+			wantPodFunc: func() corev1.Pod {
+				pod := withLabels(fakePod("foo-pod"), map[string]string{"admission.datadoghq.com/enabled": "true"})
+				pod.Spec.Containers[0].Env = append(pod.Spec.Containers[0].Env, fakeEnvWithValue("DD_ENV", "dev"))
+				pod.Spec.Containers[0].Env = append(pod.Spec.Containers[0].Env, fakeEnvWithValue("DD_SERVICE", "dd-agent"))
+				pod.Spec.Containers[0].Env = append(pod.Spec.Containers[0].Env, fakeEnvWithValue("DD_VERSION", "7"))
+				return *pod
+			},
+		},
+		{
+			name: "no labels and injectione on",
+			pod:  withLabels(fakePod("foo-pod"), map[string]string{"admission.datadoghq.com/enabled": "true"}),
+			wantPodFunc: func() corev1.Pod {
+				pod := withLabels(fakePod("foo-pod"), map[string]string{"admission.datadoghq.com/enabled": "true"})
+				return *pod
+			},
+		},
+		{
+			name: "env only and injectione on",
+			pod: withLabels(
+				fakePod("foo-pod"),
+				map[string]string{"admission.datadoghq.com/enabled": "true", "tags.datadoghq.com/env": "dev"},
+			),
+			wantPodFunc: func() corev1.Pod {
+				pod := withLabels(fakePod("foo-pod"), map[string]string{"admission.datadoghq.com/enabled": "true"})
+				pod.Spec.Containers[0].Env = append(pod.Spec.Containers[0].Env, fakeEnvWithValue("DD_ENV", "dev"))
+				return *pod
+			},
+		},
+		{
+			name: "label found but not injected",
```

Comment:
This name doesn't match the test case. We should have this test where the label isn't injected

Reply:
I change the test description a bit, but it should be right I believe. The tag label is added, but not injected because corresponding env variable is set separetly.